Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -166,6 +176,23 @@ class LoginListViewController: UIViewController {
     }
 }
 
+// MARK: - Favicon Loading
+extension LoginListViewController {
+
+    private func loadFaviconsForLogins(logins: [Login]) -> Deferred<Maybe<[GUID: [Favicon]]>> {
+        let urls = logins.map { $0.hostname }
+        return self.profile.favicons.getFaviconsForHistoryURLs(urls) >>== { favicons in
+            let filteredFavicons = favicons.asArray().flatMap { $0 }
+            var loginFaviconMap = [GUID: [Favicon]]()
+            logins.forEach { login in
+                // Filter by login hostname -> history URL and extract out the favicon
+                loginFaviconMap[login.guid] = filteredFavicons.filter { $0.0.hasPrefix(login.hostname) } .map { $0.1 } ?? []
```

Comment:
We discussed some query approaches on IRC.

This particular chunk of code won't really change much; you should still apply a transform to get the cursor into something with sub-linear lookup time.